src/common/libflux/reactor.h

```diff
@@ -91,13 +91,27 @@ flux_watcher_t *flux_buffer_read_watcher_create (flux_reactor_t *r, int fd,
 
 flux_buffer_t *flux_buffer_read_watcher_get_buffer (flux_watcher_t *w);
 
-/* 'cb' is used only for FLUX_POLLERR */
+/* 'cb' only called after fd closed (FLUX_POLLOUT) or error (FLUX_POLLERR) */
 flux_watcher_t *flux_buffer_write_watcher_create (flux_reactor_t *r, int fd,
                                                   int size, flux_watcher_f cb,
                                                   int flags, void *arg);
 
 flux_buffer_t *flux_buffer_write_watcher_get_buffer (flux_watcher_t *w);
 
+/* "write" EOF to buffer write watcher 'w'. The underlying fd will be closed
+ *  once the buffer is emptied. If data is written to the underlying
+ *  flux_buffer_t after this call, the results are undefined.
```

Comment:
"If data is written to the underlying flux_buffer_t after this call, the results are undefined." can probably be removed now.  Or if we want to, we can say attempts to write to the buffer result in EROFS.

Reply:
Good catch, thanks!